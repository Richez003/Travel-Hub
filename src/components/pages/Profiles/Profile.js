import { useSelector } from 'react-redux';
import MissionProfileCard from './MissionProfileCard';
import RocketProfileCard from './RocketProfileCard';
import './profile.css';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets);
  const missions = useSelector((state) => state.missions);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
  const joinedMissions = missions.filter((mission) => mission.reserved === true);

  return (
    <div className="profile">
      <div>
        <h2>My Rockets</h2>
        <RocketProfileCard reservedRockets={reservedRockets} />
      </div>
      <div>
        <h2>My Missions</h2>
        <MissionProfileCard joinedMissions={joinedMissions} />
      </div>
    </div>
  );
};

export default Profile;
