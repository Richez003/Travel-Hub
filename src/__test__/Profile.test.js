import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Profile from '../components/pages/Profiles/Profile';

describe('Profile Page Tests', () => {
  test('Profile Components Renders Correctly in DOM', () => {
    render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(screen.queryByText(/My Rockets/)).toBeTruthy();
    expect(screen.queryByText(/My Missions/)).toBeTruthy();
  });
  it('correctly rendered', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
