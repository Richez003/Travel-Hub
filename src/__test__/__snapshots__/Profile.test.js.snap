// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page Tests correctly rendered 1`] = `
<div>
  <div>
    <h2>
      My Rockets
    </h2>
    <table>
      <tbody />
    </table>
  </div>
  <div>
    <h2>
      My Missions
    </h2>
    <table>
      <tbody />
    </table>
  </div>
</div>
`;
